packages/common/src/util/http.js

```diff
@@ -46,19 +50,23 @@ const getAgent = (origin, { tls = {}, ...agentOpts } = {}) => {
       })
     );
 
-    agents.set(origin, agent);
+    agents.set(key, agent);
   }
 
-  return agents.get(origin);
+  return agents.get(key);
 };
 
 export const enableMockClient = (baseUrl, options = {}) => {
-  const { defaultContentType = 'application/json' } = options;
+  const {
+    defaultContentType = 'application/json',
+    ...agentOpts
+  } = options;
+  const key = `${baseUrl}|${Object.values(agentOpts).sort().join('.')}`;
```

Comment:
Key generation is now quite complicated - can we have a single function to handle this, rather than duplicating?